### PR TITLE
Replace <a> with <router-link>

### PR DESCRIPTION
**Pull request by jimmy0251**
As we are using vue router for navigation, using `<a>` breaks purpose of single page application and loads entire page again, causing performance issue. 

Let's use `<router-link>` whenever we want to navigate to internal pages.

### CODE REVIEW

File: vue-frontend/src/pages/jobs/apply/[id].vue [hunk 1]
```diff
@@ -366,28 +361,38 @@ export default {
     }),
   },
   methods: {
+    async setJob() {
+      if (this.job == null || this.job.id != this.id) {
+        await this.getCareerDetails(true);
+      }
+      this.setCareerDetails();
+    },
     async getCareerDetails() {
```
Comment (cp-sumi-k) [inline vue-frontend/src/pages/jobs/apply/[id].vue:370]: Why we haven't added `setDetails` flag here?

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: vue-frontend/src/pages/jobs/apply/[id].vue (hunk 1 above)
Comment (jimmy0251) [reply vue-frontend/src/pages/jobs/apply/[id].vue:370]: Good find!

File: vue-frontend/src/pages/jobs/apply/[id].vue [hunk 2]
```diff
@@ -366,28 +361,38 @@ export default {
...
       var req = {
         jobId: this.id,
         href: this.$route.href,
       };
-      await store.dispatch("getJobsById", req);
-      if (this.jobsError == null) {
```
Comment (cp-sumi-k) [inline vue-frontend/src/pages/jobs/apply/[id].vue:375]: Shouldn't we have to check `jobsError`?

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: vue-frontend/src/pages/jobs/apply/[id].vue (hunk 2 above)
Comment (jimmy0251) [reply vue-frontend/src/pages/jobs/apply/[id].vue:375]: No, in case of error it will go to catch block.

**Review by cp-sumi-k — APPROVED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**